topic/timeseries/requirements.txt

```diff
@@ -0,0 +1,3 @@
+plotly
+pandas>=2.0.0
+crate[sqlalchemy]
```

Comment:
There are currently problems when trying to render this notebook on Google Colab. The installation warns about `google-colab 1.0.0 requires pandas==1.5.3, but you have pandas 2.1.3 which is incompatible.`, but still installs pandas 2.x.

```
Installing collected packages: tzdata, pandas
  Attempting uninstall: pandas
    Found existing installation: pandas 1.5.3
    Uninstalling pandas-1.5.3:
      Successfully uninstalled pandas-1.5.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
lida 0.0.10 requires fastapi, which is not installed.
lida 0.0.10 requires kaleido, which is not installed.
lida 0.0.10 requires python-multipart, which is not installed.
lida 0.0.10 requires uvicorn, which is not installed.
google-colab 1.0.0 requires pandas==1.5.3, but you have pandas 2.1.3 which is incompatible.
Successfully installed pandas-2.1.3 tzdata-2023.3
```

However now, at runtime, this exception happens when using the `pandas.io.sql` subsystem together with the CrateDB driver.
```python
AttributeError: 'OptionEngine' object has no attribute 'execute'
```

![image](https://github.com/crate/cratedb-examples/assets/453543/4789113a-b765-4279-9463-3c50d02525e6)

Reply:
Another report describes the same problem, but claims it has been fixed already.

- https://github.com/huggingface/autotrain-advanced/issues/265

Reply:
Observation: I just executed 

`!pip install --upgrade pandas`

and restarted the runtime. Executing `read_sql` works afterwards. This is strange. Maybe we find a way that it works on the first try.

Reply:
It is completely natural that you observe this behavior. Code on the module scope is difficult to handle with automatic reloading mechanisms.

You can work around some details by running ad hoc code like `del sys.modules["pandas"]`, but it is always safer to restart your Jupyter kernel after modifying module scope / module import code.

Reply:
Restarting the runtime solves the issue. I did not restart after upgrading to pandas>=2.0.0.

It is not beautiful to get the compatibility error shown, but we can ignore it.

```
Requirement already satisfied: plotly in /usr/local/lib/python3.10/dist-packages (from -r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 1)) (5.15.0)
Collecting pandas>=2.0.0 (from -r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2))
  Downloading pandas-2.1.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.3/12.3 MB 20.0 MB/s eta 0:00:00
Collecting crate[sqlalchemy] (from -r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3))
  Downloading crate-0.34.0-py2.py3-none-any.whl (117 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.5/117.5 kB 13.7 MB/s eta 0:00:00
Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.10/dist-packages (from plotly->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 1)) (8.2.3)
Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from plotly->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 1)) (23.2)
Requirement already satisfied: numpy<2,>=1.22.4 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.0.0->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2)) (1.23.5)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.0.0->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2)) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.0.0->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2)) (2023.3.post1)
Collecting tzdata>=2022.1 (from pandas>=2.0.0->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2))
  Downloading tzdata-2023.3-py2.py3-none-any.whl (341 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 341.8/341.8 kB 30.2 MB/s eta 0:00:00
Requirement already satisfied: urllib3<2.1 in /usr/local/lib/python3.10/dist-packages (from crate[sqlalchemy]->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3)) (2.0.7)
Requirement already satisfied: sqlalchemy<2.1,>=1.0 in /usr/local/lib/python3.10/dist-packages (from crate[sqlalchemy]->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3)) (2.0.23)
Collecting geojson<4,>=2.5.0 (from crate[sqlalchemy]->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3))
  Downloading geojson-3.1.0-py3-none-any.whl (15 kB)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.2->pandas>=2.0.0->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 2)) (1.16.0)
Requirement already satisfied: typing-extensions>=4.2.0 in /usr/local/lib/python3.10/dist-packages (from sqlalchemy<2.1,>=1.0->crate[sqlalchemy]->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3)) (4.5.0)
Requirement already satisfied: greenlet!=0.4.17 in /usr/local/lib/python3.10/dist-packages (from sqlalchemy<2.1,>=1.0->crate[sqlalchemy]->-r https://raw.githubusercontent.com/crate/cratedb-examples/ckurze/timeseries-examples/topic/timeseries/requirements.txt (line 3)) (3.0.1)
Installing collected packages: tzdata, geojson, crate, pandas
  Attempting uninstall: pandas
    Found existing installation: pandas 1.5.3
    Uninstalling pandas-1.5.3:
      Successfully uninstalled pandas-1.5.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
lida 0.0.10 requires fastapi, which is not installed.
lida 0.0.10 requires kaleido, which is not installed.
lida 0.0.10 requires python-multipart, which is not installed.
lida 0.0.10 requires uvicorn, which is not installed.
google-colab 1.0.0 requires pandas==1.5.3, but you have pandas 2.1.3 which is incompatible.
Successfully installed crate-0.34.0 geojson-3.1.0 pandas-2.1.3 tzdata-2023.3
```